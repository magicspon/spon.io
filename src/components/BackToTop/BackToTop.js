import React from 'react'
import Hat from '@/icons/hat.inline.svg'

const BackToTop = () => (
	<button
		type="button"
		className="flex m-auto flex-col items-center justify-center"
	>
		<Hat width={20} className="pb-3" />
		<span className="text-sm text-white">Back to top</span>
	</button>
)

export default BackToTop
