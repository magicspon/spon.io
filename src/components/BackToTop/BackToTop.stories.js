import React from 'react'
import BackToTop from './BackToTop'

export default {
	title: 'BackToTop',

	parameters: {
		component: BackToTop
	}
}

export const basic = () => <BackToTop />
