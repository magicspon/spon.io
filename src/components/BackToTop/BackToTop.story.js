import React from 'react'
import { storiesOf } from '@storybook/react'
import BackToTop from './BackToTop.js'

storiesOf('BackToTop', module).add('default', () => <BackToTop />)
