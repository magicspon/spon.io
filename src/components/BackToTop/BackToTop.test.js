import React from 'react'
import { render } from 'react-testing-library'
import BackToTop from './BackToTop'

describe('BackToTop', () => {
	it('BackToTop renders correctly', () => {
		const tree = render(<BackToTop />)
		expect(tree).toMatchSnapshot()
	})
})
