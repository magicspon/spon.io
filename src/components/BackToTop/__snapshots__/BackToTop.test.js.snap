// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToTop BackToTop renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="flex m-auto flex-col items-center justify-center"
        type="button"
      >
        <svg
          class="pb-3"
          viewBox="0 0 23 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M.818 19.818h6.234l4.652-13.135h.136l4.744 13.135h6.234L14.431.182H9.227L.818 19.815z"
            fill="#35D3DB"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="text-sm text-white"
        >
          Back to top
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="flex m-auto flex-col items-center justify-center"
      type="button"
    >
      <svg
        class="pb-3"
        viewBox="0 0 23 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M.818 19.818h6.234l4.652-13.135h.136l4.744 13.135h6.234L14.431.182H9.227L.818 19.815z"
          fill="#35D3DB"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="text-sm text-white"
      >
        Back to top
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
